tests/test_regression/TestControllers.py

```diff
@@ -152,5 +152,46 @@ def test_edit_wrong_category(self):
             new_category = Category(name="C-137", description="Wubba lubba dub dub")
             g.db.add(new_category)
             g.db.commit()
-            response_regression = c.post('regression/category/1729/edit',data=dict(category_name="Sheldon", category_description="That\'s my spot", submit=True))
+            response_regression = c.post('regression/category/1729/edit',data=dict(category_name="Sheldon", category_description="Thats my spot", submit=True))
             self.assertEqual(response_regression.status_code, 404)
+
+    def test_add_test(self):
+        """
+        Check it will add a regression test
+        """
+        self.create_user_with_role(
+            self.user.name, self.user.email, self.user.password, Role.admin)
+        with self.app.test_client() as c:
+            response = c.post(
+                '/account/login', data=self.create_login_form_data(self.user.email, self.user.password))
+            response = c.post(
+                '/regression/test/new', data=dict(
+                    sample_id = 1,
+                    command = "-autoprogram -out=ttxt -latin1 -2",
+                    input_type = "file",
+                    output_type = "file",
+                    category_id = 1,
+                    expected_rc = 25,
+                    submit = True,
+                ))
+            self.assertNotEqual(RegressionTest.query.filter(RegressionTest.id==3).first(),None)
+
+    def test_add_test_empty_erc(self):
+        """
+        Check it will not add a regression test with empty Expected Runtime Code
+        """
+        self.create_user_with_role(
+            self.user.name, self.user.email, self.user.password, Role.admin)
+        with self.app.test_client() as c:
+            response = c.post(
+                '/account/login', data=self.create_login_form_data(self.user.email, self.user.password))
+            response = c.post(
+                '/regression/test/new', data=dict(
+                    sample_id = 1,
+                    command = "-autoprogram -out=ttxt -latin1 -2",
+                    input_type = InputType.file,
+                    output_type = OutputType.file,
+                    category_id = 1,
+                    submit = True,
+                ))
+            self.assertEqual(RegressionTest.query.filter(RegressionTest.id==3).first(),None)
```

Comment:
I'm missing a couple of more unittests here, with other empty fields.

Reply:
Except the command and expected_rc field all fields are dropdown so they can't be empty. And I guess command field should be allowed to be empty right? Because we can run CCX without using any addition commands as well.